Src/FluentAssertions/Primitives/ObjectAssertions.cs

```diff
@@ -16,6 +16,97 @@ public ObjectAssertions(object value)
         : base(value)
     {
     }
+
+    /// <summary>
+    /// Asserts that a value equals <paramref name="expected"/> using the provided <paramref name="comparer"/>.
+    /// </summary>
+    /// <param name="expected">The expected value</param>
+    /// <param name="comparer">
+    /// An equality comparer to compare values.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="comparer"/> is <see langword="null"/>.</exception>
+    public AndConstraint<ObjectAssertions> Be<TExpectation>(TExpectation expected, IEqualityComparer<TExpectation> comparer, string because = "", params object[] becauseArgs)
```

Comment:
We're not very consistent on this matter.
Looking in `net6.0.verified.txt` we have these stats
* expected 210
* expectation 44

I agree that "expectation" is the correct one.
The thing I like about "expected" is that the negation is "unexpected", while I'm not sure what it would be for "expectation".

Reply:
Yeah, I know we're inconsistent. Something we could fix in v7. The "problem" is that "expected" and "unexpected" are adjectives. It seems "unexpectation" is still a valid word, albeit obsolete.

Reply:
I have added a note to #1677 about considering renaming and aligning this naming.
Let's keep "expected" and "unexpected" for this API to align with the existing overloads of `Be` and `NotBe`.

Reply:
Sure. It was just a ⛏️ (nitpick) and I already approved the PR.